pkg/util/fargate/detection.go

```diff
@@ -12,14 +12,14 @@ import (
 	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
 )
 
-// IsFargateInstance returns whether the Agent is running in a sidecar deployment mode.
+// IsSidecar returns whether the Agent is running in a sidecar deployment mode.
```

Comment:
Seems like this could be moved out of the `fargate` package if it's not specifically for Fargate anymore